File: app/src/main/java/net/xzos/UpgradeAll/server/app/engine/js/utils/JSLog.java
package net.xzos.UpgradeAll.server.app.engine.js.utils;

import net.xzos.UpgradeAll.server.ServerContainer;
import net.xzos.UpgradeAll.server.log.LogUtil;

public class JSLog {

    protected static final LogUtil Log = ServerContainer.AppServer.getLog();

    private String[] LogObjectTag;
    private String TAG = "JavaScriptRunning";

    public JSLog(String[] logObjectTag) {
        this.LogObjectTag = logObjectTag;
    }

    public void v(Object msgObject) {
        Log.v(LogObjectTag, TAG, msgObject);
    }

    public void d(Object msgObject) {
        Log.d(LogObjectTag, TAG, msgObject);
    }

    public void i(Object msgObject) {
        Log.i(LogObjectTag, TAG, msgObject);
    }

    public void w(Object msgObject) {
        Log.w(LogObjectTag, TAG, msgObject);
    }

    public void e(Object msgObject) {
        Log.e(LogObjectTag, TAG, msgObject);
    }
}

File: app/src/main/kotlin/net/xzos/upgradeAll/server/app/engine/js/utils/JSLog.kt
package net.xzos.upgradeAll.server.app.engine.js.utils

import net.xzos.upgradeAll.server.ServerContainer

class JSLog(private val LogObjectTag: Array<String>) {

    fun v(msg: String) {
        Log.v(LogObjectTag, TAG, msg)
    }

    fun d(msg: String) {
        Log.d(LogObjectTag, TAG, msg)
    }

    fun i(msg: String) {
        Log.i(LogObjectTag, TAG, msg)
    }

    fun w(msg: String) {
        Log.w(LogObjectTag, TAG, msg)
    }

    fun e(msg: String) {
        Log.e(LogObjectTag, TAG, msg)
    }

    companion object {
        private val Log = ServerContainer.Log
        private const val TAG = "JavaScriptRunning"
    }
}
